test/cpp/api/tensor.cpp

```diff
@@ -219,20 +277,84 @@ TEST(TensorTest, ContainsCorrectValuesForManyValuesVariable) {
   tensor = torch::tensor(std::vector<double>({1.5, 2.25, 3.125}));
   ASSERT_TRUE(tensor.is_variable());
   ASSERT_EQ(tensor.numel(), 3);
+  ASSERT_EQ(tensor.sizes(), std::vector<int64_t>({3}));
   ASSERT_EQ(tensor.dtype(), at::kDouble);
   ASSERT_TRUE(almost_equal(tensor[0], 1.5));
   ASSERT_TRUE(almost_equal(tensor[1], 2.25));
   ASSERT_TRUE(almost_equal(tensor[2], 3.125));
+
+  tensor = torch::tensor({true, false, true});
+  ASSERT_TRUE(tensor.is_variable());
+  ASSERT_EQ(tensor.numel(), 3);
+  ASSERT_EQ(tensor.sizes(), std::vector<int64_t>({3}));
+  ASSERT_EQ(tensor.dtype(), at::kBool);
+  ASSERT_TRUE(exactly_equal(tensor[0], true));
+  ASSERT_TRUE(exactly_equal(tensor[1], false));
+  ASSERT_TRUE(exactly_equal(tensor[2], true));
+
+  tensor = torch::tensor(at::ArrayRef<bool>({true, false, true}));
+  ASSERT_TRUE(tensor.is_variable());
+  ASSERT_EQ(tensor.numel(), 3);
+  ASSERT_EQ(tensor.sizes(), std::vector<int64_t>({3}));
+  ASSERT_EQ(tensor.dtype(), at::kBool);
+  ASSERT_TRUE(exactly_equal(tensor[0], true));
+  ASSERT_TRUE(exactly_equal(tensor[1], false));
+  ASSERT_TRUE(exactly_equal(tensor[2], true));
 }
 
 TEST(TensorTest, MultidimTensorCtor) {
+  {
```

Comment:
Can we split this test into smaller tests?

i.e., we could have a "zero-sized dim test" which are all of the {} / {{},{}} ones

Also, it might be good to assert the dtype of torch::tensor({}) (I'm not sure if that test exists already)

Reply:
Added `TensorCtorZeroSizedDim` and also fixed the code to let tensor with zero-sized dim to be of `float` dtype by default, matching the Python API behavior.